File: Ander systeem/Core.ApplicationServices/DTO/ModelingProcess/ModelStorageDTO.cs
using Leap.Domain.Domain.ModelConfig.Enums;

namespace Leap.ApplicationServices.DTO.ModelingProcess
{
    /// <summary>
    /// Functions as the request body for persisting
    /// </summary>
    public class ModelStorageCreationRequestDTO
    {
        public required Guid WorkspaceGuid { get; set; }
        public required Guid DataExtractConfigGuid { get; set; }
        public required Guid ModelConfigGuid { get; set; }
        public bool Overwrite { get; set; } = false;
    }


    /// <summary>
    /// Functions as the response dto after modelstorage was succesfull.
    /// 
    /// Contents:
    /// - WorkspaceId
    /// - Unique address of model location
    /// - Name of the model
    /// - Date of creation
    /// - Version
    /// </summary>
    public class ModelStorageDTO : IDTO
    {
        public required Guid WorkspaceGuid { get; set; }
        public required string ModelAddress { get; set; }
        public required string ModelName { get; set; }
        public required string ModelVersion { get; set; } = "latest";
        public required ModelType ModelType { get; set; }
        public required ModelAlgorithm ModelAlgorithm { get; set; }
        public required DateTime DateOfCreation { get; set; }
    }
}

File: App Python/Domain/DTO/ModelStorageDTO.py
from datetime import datetime, timezone

from pydantic import BaseModel

from Domain import DataRequestDTO, ModelAlgorithm, ModelConfigDTO, ModelType

from .CalculationStepDTO import CalculationStepDTO


class ModelStorageCreationDTO(BaseModel):
    DateOfAction: datetime
    DataRequest: DataRequestDTO
    ModelConfig: ModelConfigDTO
    OperationList: list[CalculationStepDTO] = []
    

class ModelStorageDTO(BaseModel):
    """
    This DTO will be used as a response object to the client
    that contains all of the meta data
    """
    WorkspaceGuid: str 
    ModelAddress: str 
    ModelName: str 
    ModelVersion: str = "latest"
    ModelType: ModelType
    ModelAlgorithm: ModelAlgorithm
    DateOfCreation: datetime = datetime.now(timezone.utc)